### PR TITLE
Dyn 3908 open controls by step

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

I modified the guides json file to include the UI Automation part
Also I created a new class called StepUIAutomation that will be used by a Step to fetch the information from the json file.
I modified the GuidesManager class for including a method that will create the Automation information and the Guide class for UIAutomation action that will be used when the Next or Back button are pressed in the current Popup.
Finally I added some event handlers for the RealTimeInfoWindow, (this popup will appear just when the current Tour is closed without finishing it). This event handlers will prevent to have a pile of RealTimeInfoWindow if several tours are not finished.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang 

### FYIs

@Astul-Betizagasti

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
This is a GIF of how Dynamo looks like when opening the menues.
guided_tour_openmenues

File: src/DynamoCoreWpf/UI/GuidedTour/Guide.cs [hunk 1]
```diff
@@ -106,12 +107,29 @@ public void Next(GuidedTourMovementEventArgs args)
...
 
             CurrentStep = (from step in GuideSteps where step.Sequence == args.StepSequence select step).FirstOrDefault();
             if (CurrentStep != null)
+            {
                 CurrentStep.Hide();
+
+                //Disable the current action automation that is executed for the Current Step (if there is one)
+                if (CurrentStep.UIAutomation != null)
+                {
+                    ExecuteUIAutomationStep(CurrentStep.UIAutomation, false);
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/Guide.cs:130]: can we remove the added code for nextStep and simply execute UIAutomation for each `current step`? I feel that is easier to understand and implement but not sure if there is other limitation

File: src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs [hunk 2]
```diff
@@ -179,6 +184,29 @@ private HostControlInfo CreateHostControl(Step jsonStepInfo)
             return popupInfo;
         }
 
+        /// <summary>
+        /// This methos will create an StepUIAutomation instance based in the information passed as parameter
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs:188]: `based on`

**Review by QilongTang — COMMENTED** [review]
A few comments then LGTM

**Comment by QilongTang** [comment]
I also noticed the text is not exactly same with what was defined on this wiki, maybe we were still referencing figma design? Please update according to https://wiki.autodesk.com/display/aeceng/Content+for+Dynamo+First+Time+Use+Guide

File: src/DynamoCoreWpf/UI/GuidedTour/Guide.cs (hunk 1 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/UI/GuidedTour/Guide.cs]: the code added to NextStep() and PrevStep() were moved from Guide to Step class, so every time that the Step.Show() or Step.Hide() methods are called the UI Automation will be executed for hiding or showing the functionality.
see commit: https://github.com/DynamoDS/Dynamo/pull/11937/commits/65d76d3883ad2d050273a178016f454345b30ea8

File: src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs (hunk 2 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs]: @QilongTang this comment was updated in the next commit:  https://github.com/DynamoDS/Dynamo/pull/11937/commits/65d76d3883ad2d050273a178016f454345b30ea8

**Comment by RobertGlobant20** [comment]
> I also noticed the text is not exactly same with what was defined on this wiki, maybe we were still referencing figma design? Please update according to https://wiki.autodesk.com/display/aeceng/Content+for+Dynamo+First+Time+Use+Guide

@QilongTang all the content for each step was updated accordingly to the wiki entry Content for Dynamo First Time Use Guide.

**Comment by RobertGlobant20** [comment]
GIF updated according to the changes in the popups content
guided_tour_openmenues2

**Review by QilongTang — APPROVED** [review]

**Pull request merged**